src/main/java/seedu/address/model/animal/KeywordPredicate.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.model.animal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Animal}'s {@code Name} matches any of the keywords given.
+ */
+public class KeywordPredicate implements Predicate<Animal> {
+    private final List<String> keywords;
+
+    public KeywordPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Animal animal) {
+
+        List<String> stringAttributes = new ArrayList<>(List.of(
+                animal.getNameForSerialization(),
+                animal.getPetIdForSerialization(),
+                animal.getDateOfBirthForSerialization(),
+                animal.getAdmissionDateForSerialization(),
+                animal.getSpeciesForSerialization(),
+                animal.getSexForSerialization(),
+                animal.getBreedForSerialization()
+        ));
+
+        assert(keywords.size() == stringAttributes.size());
```

Comment:
Do add an inline comment as to why this assertion should hold true, or something like
`assert (condition) : "what went wrong"`